### PR TITLE
Neural Search step type

### PR DESCRIPTION
**Pull request by JWittmeyer**
main PR: https://github.com/code-kern-ai/cognition-gateway/pull/122

### CODE REVIEW

File: business_objects/embedding.py [hunk 1]
```diff
@@ -327,6 +340,74 @@ def get_match_record_ids_to_qdrant_ids(
     return [r[0] for r in general.execute_all(query)]
 
 
+def get_match_record_ids_to_qdrant_ids_with_max_score(
+    project_id: str, embedding_id: str, qdrant_results: List[Any], limit: int
+) -> List[Any]:
```
Comment (lumburovskalina) [inline business_objects/embedding.py:345]: not sure if it makes sense but usually we have a default value for limit?

File: business_objects/embedding.py [hunk 2]
```diff
@@ -327,6 +340,74 @@ def get_match_record_ids_to_qdrant_ids(
...
+    GROUP BY et.record_id
+    ORDER BY 2 ASC
+    LIMIT {limit}
+    """
+
+    return [{"id": r[0], "score": r[1]} for r in general.execute_all(query)]
+
+
+def __generate_with_table_union_query(qdrant_results: List[Any]) -> str:
+
```
Comment (lumburovskalina) [inline business_objects/embedding.py:393]: prevent_sql_injection for qdrant_results?

File: business_objects/embedding.py (hunk 1 above)
Comment (JWittmeyer) [reply business_objects/embedding.py:345]: not in this case as it's a required parameter calculated by the embedding or neural search service

File: business_objects/embedding.py (hunk 2 above)
Comment (JWittmeyer) [reply business_objects/embedding.py:393]: not needed as the values aren't from outside & not string based but a return object class from qdrant itself.

**Review by lumburovskalina — APPROVED** [review]

**Pull request merged**